src/internal/group_api/mod.rs

```diff
@@ -359,88 +372,153 @@ pub(crate) fn get_group_keys<'a>(
-                            _,
-                        > = member_ids_and_keys
-                            .into_iter()
-                            .map(|(id, user_pub_key)| {
-                                let maybe_transform_key = recrypt.generate_transform_key(
-                                    &group_priv_key.clone().into(),
-                                    &user_pub_key.clone().into(),
-                                    &auth.signing_private_key().into(),
-                                );
-                                maybe_transform_key.map(|transform_key| {
-                                    (id, (user_pub_key, Some(transform_key.into())))
-                                })
+    use futures::future::Either;
+    let owner_id = match owner.clone() {
+        Some(id) => id,
+        None => auth.account_id().clone(),
+    };
+    // don't make any requests if validation fails
+    if admins.is_empty() {
+        Either::A(err(IronOxideErr::ValidationError(
+            format!("admins"),
+            format!("admins list cannot be empty"),
+        )))
+    } else if !admins.contains(&owner_id) {
+        Either::A(err(IronOxideErr::ValidationError(
+            format!("admins"),
+            format!("admins list must contain the owner"),
+        )))
+    } else {
+        Either::B(
+            user_api::user_key_list(auth, users_to_lookup)
+                .and_then(move |user_ids_and_keys| {
+                    // this will occur when one of the UserIds cannot be found
+                    if user_ids_and_keys.len() != users_to_lookup.len() {
+                        // figure out which user ids could not be found in the database and include them in the error message.
+                        let desired_users_set: HashSet<UserId> =
+                            HashSet::from_iter(users_to_lookup.clone());
+                        let found_users_set: HashSet<UserId> =
+                            user_ids_and_keys.into_iter().map(|(x, _)| x).collect();
+                        let diff: HashSet<&UserId> =
+                            desired_users_set.difference(&found_users_set).collect();
+                        err(IronOxideErr::UserDoesNotExist(format!(
+                            "Failed to find the following users: {:?}",
+                            diff
+                        )))
+                    } else {
+                        transform::gen_group_keys(recrypt)
+                            .and_then(move |(plaintext, group_priv_key, group_pub_key)| {
+                                let (member_info_result, admin_info): (
+                                    Result<Vec<(UserId, PublicKey, TransformKey)>, IronOxideErr>,
+                                    Vec<(UserId, PublicKey)>,
+                                ) = {
+                                    let members_set: HashSet<_> = HashSet::from_iter(&members);
```

Comment:
I factored this out into `collect_admin_and_member_info()`, but it takes so many different things (because it generates transform keys for each member) that I'm having trouble including a test for this specifically.

Reply:
Okay, I was able to make tests on this function, but I'm not sure what good test cases would be. I just have a test with filled admins and members lists, and a test with an empty members list. The admins list will have been validated to never be empty ahead of time.

Reply:
Consider using https://github.com/dac-gmbh/vec1 for NonEmptyList and get rid of error cases at compile time :)

We want to bring this in anyway...
https://github.com/IronCoreLabs/ironoxide/issues/18#issuecomment-519352993

Reply:
I made `admins` a NonEmptyList in `GroupCreateOptsStd` and throughout the `group_create` calls